### PR TITLE
Create and Design the Show and Edit views for Curriculum

### PR DESCRIPTION
**Pull request by eduardotru**
Create and design the Show and Edit view for the Curriculum of a student.
Added cocoon gem to make it simpler to have repeated nested attributes.
Also fixed a typo in the migrations.

## Show View
Screenshot from 2020-04-29 23-10-00
Screenshot from 2020-04-29 23-10-05

# Edit View
Screenshot from 2020-04-29 23-10-13
Screenshot from 2020-04-29 23-10-19
Screenshot from 2020-04-29 23-10-22

### CODE REVIEW

File: app/views/curriculums/_certification_fields.html.erb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      <div class="col s12 l2">
+        <%= f.label :Fecha %>
+        <%= f.date_field :date_emitted %>
+      </div>
+    </div>
+
+    <div class="row">
+      <div class="col s12">
+        <%= f.label :Descripcion_breve %>
+        <%= f.text_area :brief_description %>
```
Comment (ReneSG) [inline app/views/curriculums/_certification_fields.html.erb:24]: Can you add the css class to style the text are with materialize. There are some examples in our codebase.

File: app/views/curriculums/_education_history_fields.html.erb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      <div class="col s6 l2">
+        <%= f.label :Fin_de_estudios %>
+        <%= f.date_field :end_date %>
+      </div>
+    </div>
+
+    <div class="row">
+      <div class="col s12">
+        <%= f.label :Descripcion_breve %>
+        <%= f.text_area :brief_description %>
```
Comment (ReneSG) [inline app/views/curriculums/_education_history_fields.html.erb:28]: ditto

File: app/views/curriculums/_form.html.erb [hunk 3]
```diff
@@ -1,2 +1,110 @@
...
+  <div class="col s12 l3">
+    <%= f.label :Maximo_salario_esperado %>
+    <%= f.number_field :max_desired_salary %>
+  </div>
+</div>
+
+<blockquote>
+  <h5>Objetivo profesional:</h5>
+  <%= f.label :Objetivo_Profesional %>
+  <%= f.text_area :professional_objective %>
```
Comment (ReneSG) [inline app/views/curriculums/_form.html.erb:27]: Ditto.

File: app/views/curriculums/_form.html.erb [hunk 4]
```diff
@@ -1,2 +1,110 @@
...
+      <div class="row">
+        <div class="col s12">
+          <h5>Habilidades:</h5>
+        </div>
+      </div>
+      <div class="card-panel">
+        <div class="row">
+          <div class="col s12">
+            <%= f.label :Habilidades %>
+            <%= f.text_area :skills %>
```
Comment (ReneSG) [inline app/views/curriculums/_form.html.erb:63]: ditto

File: app/views/curriculums/_form.html.erb [hunk 5]
```diff
@@ -1,2 +1,110 @@
...
+      <div class="row">
+        <div class="col s12">
+          <h5>Áreas de expertiz:</h5>
+        </div>
+      </div>
+      <div class="card-panel">
+        <div class="row">
+          <div class="col s12">    
+            <%= f.label :Areas_de_expertiz %>
+            <%= f.text_area :areas_of_expertise %>
```
Comment (ReneSG) [inline app/views/curriculums/_form.html.erb:80]: Ditto

File: app/views/curriculums/_form.html.erb [hunk 6]
```diff
@@ -1,2 +1,110 @@
-<%= f.label :Objetivo_Profesional %>
-<%= f.text_area :professional_objective %>
\ No newline at end of file
+<div class="row">
+  <div class="col s12">
+    <h5>Foto:</h5>
+    <%= f.file_field :profile_picture %>
```
Comment (ReneSG) [inline app/views/curriculums/_form.html.erb:4]: Please add the css class for materialize file input styles. There is an example in `company/edit`

File: app/views/curriculums/_language_fields.html.erb [hunk 7]
```diff
@@ -0,0 +1,24 @@
+<div class='nested-fields'>
+  <div class="row">
+    <div class="col s4">
+      <%= f.label :Idioma %>
+      <%= f.text_field :language_name %>
+    </div>
+    <div class="col s6">
+      <%= f.label :Nivel_de_conocimiento %>
+      <%= f.select :proficiency, options_for_select([
```
Comment (ReneSG) [inline app/views/curriculums/_language_fields.html.erb:9]: Add styles for materialize dropdown.

Comment (ReneSG) [inline app/views/curriculums/_language_fields.html.erb:9]: Add styles for materialize dropdown.

File: app/views/curriculums/_form.html.erb (hunk 6 above)
Comment (ReneSG) [inline app/views/curriculums/_form.html.erb:4]: Add styles for materialize file input. There is an example in `company/edit`

File: app/views/curriculums/_certification_fields.html.erb (hunk 1 above)
Comment (ReneSG) [inline app/views/curriculums/_certification_fields.html.erb:24]: Add styles for materialize text_area. There are some examples in our codebase.

File: app/views/curriculums/_education_history_fields.html.erb (hunk 2 above)
Comment (ReneSG) [inline app/views/curriculums/_education_history_fields.html.erb:28]: Ditto.

File: app/views/curriculums/_form.html.erb (hunk 3 above)
Comment (ReneSG) [inline app/views/curriculums/_form.html.erb:27]: Ditto.

File: app/views/curriculums/_form.html.erb (hunk 4 above)
Comment (ReneSG) [inline app/views/curriculums/_form.html.erb:63]: Ditto.

File: app/views/curriculums/_form.html.erb (hunk 5 above)
Comment (ReneSG) [inline app/views/curriculums/_form.html.erb:80]: Ditto.

File: app/assets/stylesheets/application.scss [hunk 8]
```diff
@@ -130,3 +130,26 @@
   color: black;
 }
 
+.zero-margin {
```
Comment (ReneSG) [inline app/assets/stylesheets/application.scss:133]: Can we use camelcase for the class names. I think that it was the style we were using across all the codebase. Same for all below.

File: app/views/curriculums/_language_fields.html.erb (hunk 7 above)
Comment (eduardotru) [reply app/views/curriculums/_language_fields.html.erb:9]: Duplicated comment

Comment (eduardotru) [reply app/views/curriculums/_language_fields.html.erb:9]: Materialize select is not being used in the whole app and it seems it's really buggy, you can see it has `browser-default` class in every select.

**Comment by eduardotru** [comment]
Resolved all the comments, here is how the edit page looks now:

image

**Pull request merged**